packages/graphql-language-service-server/src/findGraphQLTags.js

```diff
@@ -0,0 +1,186 @@
+/**
+ *  Copyright (c) Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the license found in the
+ *  LICENSE file in the root directory of this source tree.
+ *
+ *  @flow
+ */
+
+import {Position, Range} from 'graphql-language-service-utils';
+
+import {parse} from 'babylon';
+
+// Attempt to be as inclusive as possible of source text.
+const BABYLON_OPTIONS = {
+  allowImportExportEverywhere: true,
+  allowReturnOutsideFunction: true,
+  allowSuperOutsideMethod: true,
+  sourceType: 'module',
+  plugins: [
+    // Previously "*"
+    'asyncGenerators',
+    'classProperties',
+    'decorators',
+    'doExpressions',
+    'dynamicImport',
+    'exportExtensions',
+    'flow',
+    'functionBind',
+    'functionSent',
+    'jsx',
+    'objectRestSpread',
+  ],
+  strictMode: false,
+};
+
+export function findGraphQLTags(
+  text: string,
+  filePath: string,
+): Array<{tag: string, template: string, range: Range}> {
+  const result = [];
+  const ast = parse(text, BABYLON_OPTIONS);
+
+  const visitors = {
+    CallExpression: node => {
+      const callee = node.callee;
+      if (
+        !((callee.type === 'Identifier' &&
+          CREATE_CONTAINER_FUNCTIONS[callee.name]) ||
+          (callee.kind === 'MemberExpression' &&
+            callee.object.type === 'Identifier' &&
+            callee.object.value === 'Relay' &&
+            callee.property.type === 'Identifier' &&
+            CREATE_CONTAINER_FUNCTIONS[callee.property.name]))
+      ) {
+        traverse(node, visitors);
+        return;
+      }
+      const fragments = node.arguments[1];
+      if (fragments.type === 'ObjectExpression') {
+        fragments.properties.forEach(property => {
+          const tagName = getGraphQLTagName(property.value.tag);
+          const template = getGraphQLText(property.value.quasi);
+          if (tagName === 'graphql' || tagName === 'graphql.experimental') {
+            const loc = property.value.loc;
+            const range = new Range(
```

Comment:
This is where I compute the range of the quasi - needed in the language service to locate the query string and compute the range for, i.e. the range of diagnostic results.